extensions/amp-sidebar/0.1/amp-sidebar.js

```diff
@@ -0,0 +1,254 @@
+      dev.warn(TAG, `Experiment ${EXPERIMENT} disabled`);
+      return;
+    }
+    if (!this.checkWhitelist_()) {
+      return;
+    }
+
+    if (this.isIosSafari_) {
+      this.fixIosElasticScrollLeak_();
+    }
+
+    this.adjustPadding_();
+
+    if (this.documentElement_.classList.contains('amp-sidebar-open')) {
+      // Create the mask if the sidebar is rendered in open mode.
+      this.createMask_();
+    }
+
+    this.documentElement_.addEventListener('keydown', event => {
+      // Close sidebar on ESC.
+      if (event.keyCode == 27) {
+        this.close_();
+      }
+    });
+    //TODO (skrish, #2712) Add history support on back button.
+    this.registerAction('activate', this.open_.bind(this));
+    this.registerAction('toggle', this.toggle_.bind(this));
+    this.registerAction('open', this.open_.bind(this));
+    this.registerAction('close', this.close_.bind(this));
+  }
+
+  /**
+   * @private
+   */
+  adjustPadding_() {
+    const viewerPaddingTop = this.viewport_.getPaddingTop();
+    if (viewerPaddingTop) {
+      // viewerPaddingTop exists when AMP page is rendered inside search
+      // carousel.
+      const div = this.document_.createElement('div');
+      setStyles(div, {
+        'height': viewerPaddingTop + 'px',
+        'width': '100%',
+      });
+      const firstChild = this.element.firstChild;
+      this.element.insertBefore(div, firstChild);
+    }
+    if (this.isIosSafari_) {
+      //Compensate for IOS safari bottom navbar.
+      const div = this.document_.createElement('div');
+      setStyles(div, {
+        'height': IOS_SAFARI_BOTTOMBAR_HEIGHT,
+        'width': '100%',
+      });
+      this.element.appendChild(div);
+    }
+  }
+
+  /**
+   * Toggles the open/close state of the sidebar.
+   * @private
+   */
+  toggle_() {
+    if (this.documentElement_.classList.contains('amp-sidebar-open')) {
+      this.close_();
+    } else {
+      this.open_();
+    }
+  }
+
+  /**
+   * Reveals the sidebar.
+   * @private
+   */
+  open_() {
+    this.mutateElement(() => {
+      this.getViewport().addToFixedLayer(this.element);
+      this.createMask_();
+      this.viewport_.disableTouchZoom();
+      this.documentElement_.classList.add('amp-sidebar-open');
+      this.documentElement_.classList.remove('amp-sidebar-closed');
+      this.element./*REVIEW*/scrollTop = 1;
+    });
+  }
+
+  /**
+   * Hides the sidebar.
+   * @private
+   */
+  close_() {
+    this.mutateElement(() => {
+      this.documentElement_.classList.remove('amp-sidebar-open');
+      this.documentElement_.classList.add('amp-sidebar-closed');
+      this.getViewport().removeFromFixedLayer(this.element);
+      this.getViewport().restoreOriginalTouchZoom();
+    });
+  }
+
+  /**
+   * Checks if the sidebar only has the whitlisted custom amp- elements.
+   * @returns {boolean} True when only whitelited elements are present.
+   * @private
+   */
+  checkWhitelist_() {
+    const elements = this.element.getElementsByTagName('*');
```

Comment:
ads - iframes , instagram, FB , videos , Youtube etc ... its a long list. 

I would assume that there will be a limited no of elements on the sidebar (considering the space available). We could possibly remove this if/when this gets implemented in validation.